plugins/mux/serve_interceptor.go

```diff
@@ -58,8 +61,23 @@ func (n *ServeHTTPInterceptor) AfterInvoke(invocation operator.Invocation, resul
 	return nil
 }
 
+func newWriterWrapper(rw interface{}) (*writerWrapper, error) {
+	writer := rw.(http.ResponseWriter)
+	hijacker, ok := rw.(http.Hijacker)
+	if !ok {
+		return nil, fmt.Errorf("http.ResponseWriter does not implement http.Hijacker")
```

Comment:
If cast failure, could you align with the older strategy? We should not break other framework use.

Reply:
Set hijacker to nil? Or implement the `Hijack` method ？

Reply:
I think the original wrapper has already implement the hijick interface? Right? I don't know why you add it again.

Reply:
Replaced original response writer with `writerWrapper` in `BeforeInvoke`

Reply:
You mean the framework not work? Then I think you need to set it as nil if the cast check failure. And if the cast success, you need to call the original hijack method. Right? Otherwise, your framework will still failure for now.

Reply:
Some websocket server impl:

https://github.com/gorilla/websocket/blob/main/server.go#L175
https://github.com/lxzan/gws/blob/main/upgrader.go#L93

Reply:
也许这里可以使用 `http.ResponseController`

Reply:
https://github.com/golang/go/blob/master/src/net/http/responsecontroller.go#L66

Reply:
I mean, you cannot break the original plugin. The hijack is only an optional, not requirement.

Reply:
Sorry, using `http.ResponseController` this object needs to be in 1.20, now i just print log.

Reply:
A simple way is to let the wrapper implement the Hijack interface, and If the original writer implements the Hijack, you can call the original method. 
Another way is you can create two wrappers, if the case fails, use the current code. otherwise, you create a new write with the Hijack interface.

Reply:
I tried to add some code, which introduced the `bufio` and `net` packages, and then the following error occurred when building. How to solve it?

```
$WORK/b289/skywalking_enhance_mux_serve_interceptor.go:21:2: could not import bufio (open : no such file or directory)
$WORK/b289/skywalking_enhance_mux_serve_interceptor.go:23:2: could not import net (open : no such file or directory)
```

Reply:
You cannot import the package which the destination package is imported.

Reply:
..., Hijack signature must introduce these two packages.
```go
func (w *writerWrapper) Hijack() (net.Conn, *bufio.ReadWriter, error) {
	rw := w.ResponseWriter
	switch t := rw.(type) {
	case http.Hijacker:
		return t.Hijack()
	default:
		return nil, nil, http.ErrNotSupported
	}
}

```

Reply:
Then, you can try the second resolution what I mentioned before.